src/subscription.ts

```diff
@@ -0,0 +1,42 @@
+import * as extensionApi from "@podman-desktop/api";
+import { accessSync, constants, readFileSync } from "node:fs";
+import { homedir } from "node:os";
+import path from "node:path";
+
+export const REGISTRY_REDHAT_IO = 'registry.redhat.io';
+
+export async function signIntoRedHatDeveloperAccount(
+  createIfNone = true
+): Promise<extensionApi.AuthenticationSession | undefined> {
+  return extensionApi.authentication.getSession(
+    'redhat.authentication-provider',
+    [
+      'api.iam.registry_service_accounts', //scope that gives access to hydra service accounts API
+      'api.console',
+    ], // scope that gives access to console.redhat.com APIs
+    { createIfNone }
+  );
+}
+
+// TODO: add listRegistries to registry API to allow search by
+// registry URL
```

Comment:
listRegistries might be async so I I don't know if your function is synchronous if it'll work as expected (you'll need some refactoring)

Reply:
I was wondering if you can use `onDidRegisterRegistry` method to collect registries ?

Reply:
it should notify with all podman registries

Reply:
Those are all good options with pros and cons. Let's take care of it as follow up issue.